From 348c4985246d76f762c3d9a96a3b961a53730518 Mon Sep 17 00:00:00 2001
From: peijiankang <peijiankang@kylinos.cn>
Date: Wed, 22 Mar 2023 14:16:27 +0800
Subject: [PATCH] fix version of kylin-usb-creator

---
 UIControl/src/include/menumodule.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/UIControl/src/include/menumodule.h b/UIControl/src/include/menumodule.h
index ca42c31..d9deeb2 100644
--- a/UIControl/src/include/menumodule.h
+++ b/UIControl/src/include/menumodule.h
@@ -56,7 +56,7 @@ public:
 public:
 //    程序在实例化的时候需要传递的信息字段,打开debug开关后这些字段会被自动填充
     QString appName = "tools/kylin-usb-creator"; //格式kylin-usb-creator
-    QString appVersion = "1.1.0";
+    QString appVersion = "1.1.1";
     QString appDesc = "appDesc字段未填充!";
     QString iconPath = ":/data/kylin-usb-creator.svg";
     QString confPath = "org.kylin-usb-creator-data.settings";
-- 
2.39.1

